Benchmark/System/Process.hs

```diff
@@ -25,84 +21,90 @@ import qualified Streamly.Data.Fold as FL
+-- Create a data file filled with random data
+-------------------------------------------------------------------------------
 
 ddBlockSize :: Int
-ddBlockSize = 1024
+ddBlockSize = 1024 * 1024
 
 ddBlockCount :: Int
-ddBlockCount = 100              -- ~100 KB
-
-numCharInCharFile :: Int
-numCharInCharFile = 100 * 1024  -- ~100 KB
-
-ioCatBinary :: IO FilePath
-ioCatBinary = do
-    maybeDdBinary <- findExecutable "cat"
-    case maybeDdBinary of
-        Just ddBinary -> return ddBinary
-        _ -> error "cat Binary Not Found"
-
-ioTrBinary :: IO FilePath
-ioTrBinary = do
-    maybeDdBinary <- findExecutable "tr"
-    case maybeDdBinary of
-        Just ddBinary -> return ddBinary
-        _ -> error "tr Binary Not Found"
+ddBlockCount = 10
 
 largeByteFile :: String
 largeByteFile = "./largeByteFile"
 
+generateByteFile :: IO ()
+generateByteFile = do
+    ddPath <- which "dd"
+    let procObj = proc ddPath [
+                "if=" ++ devRandom,
+                "of=" ++ largeByteFile,
+                "count=" ++ show ddBlockCount,
+                "bs=" ++ show ddBlockSize
+            ]
+
+    (_, _, _, procHandle) <- createProcess procObj
+    _ <- waitForProcess procHandle
+    return ()
+
+-------------------------------------------------------------------------------
+-- Create a file filled with ascii chars
+-------------------------------------------------------------------------------
+
 largeCharFile :: String
 largeCharFile = "./largeCharFile"
 
-executableFile :: String
-executableFile = "./writeTrToError.sh"
-
-executableFileContent :: String
-executableFileContent =
-    "tr [a-z] [A-Z] <&0 >&2"
-
-createExecutable :: IO ()
-createExecutable = do
-    writeFile executableFile executableFileContent
-    callCommand ("chmod +x " ++ executableFile)
-
-generateByteFile :: IO ()
-generateByteFile =
-    do
-        ddBinary <- ioDdBinary
-        let procObj = proc ddBinary [
-                    "if=" ++ devRandom,
-                    "of=" ++ largeByteFile,
-                    "count=" ++ show ddBlockCount,
-                    "bs=" ++ show ddBlockSize
-                ]
-
-        (_, _, _, procHandle) <- createProcess procObj
-        _ <- waitForProcess procHandle
-        return ()
+numCharInCharFile :: Int
+numCharInCharFile = 10 * 1024 * 1024
 
 generateCharFile :: IO ()
 generateCharFile = do
     handle <- openFile largeCharFile WriteMode
     FH.putBytes handle (S.replicate numCharInCharFile _a)
     hClose handle
 
+-------------------------------------------------------------------------------
+-- Create a utility that writes to stderr
+-------------------------------------------------------------------------------
+
+trToStderr :: String
+trToStderr = "./writeTrToError.sh"
+
+trToStderrContent :: String
+trToStderrContent =
+    "tr [a-z] [A-Z] <&0 >&2"
+
+createExecutable :: IO ()
+createExecutable = do
+    writeFile trToStderr trToStderrContent
+    callCommand ("chmod +x " ++ trToStderr)
```

Comment:
Why are we creating and deleting files from the Haskell code? To keep it all in one place?

Reply:
This is how it was written by the original author. I think we can create a script file and commit it.